ui/src/app/views/bug-report/component.tsx

```diff
@@ -0,0 +1,139 @@
+import {Component, Input} from '@angular/core';
+
+import {bugReportApi} from 'app/services/swagger-fetch-clients';
+import {reactStyles, ReactWrapperBase, withUserProfile} from 'app/utils/index';
+
+import {AlertDanger} from 'app/components/alert';
+import {Button} from 'app/components/buttons';
+import {ClrIcon} from 'app/components/icons';
+import {Modal, ModalBody, ModalFooter, ModalInput, ModalTitle} from 'app/components/modals';
+import {
+  BugReport,
+  BugReportType,
+  Profile
+} from 'generated/fetch';
+import * as React from 'react';
+
+
+const styles = reactStyles({
+  fieldHeader: {
+    fontSize: 14,
+    color: '#262262',
+    fontWeight: 200,
+    display: 'block',
+    marginTop: '1rem'
+  },
+  field: {
+    fontWeight: 400,
+    width: '100%'
+  }
+});
+
+export const BugReportModal = withUserProfile()
+(class extends React.Component<
+    {profileState: { profile: Profile, reload: Function },
+      bugReportDescription: string, onClose: Function},
+    {bugReport: BugReport, submitting: boolean,
+      sendBugReportError: boolean}> {
+
+  private emptyBugReport = {
+    bugType: BugReportType.APPLICATION,
+    shortDescription: '',
+    reproSteps: '',
+    includeNotebookLogs: true,
+    contactEmail: ''
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      submitting: false,
+      sendBugReportError: false,
+      bugReport: {...this.emptyBugReport,
+        shortDescription: props.bugReportDescription || '',
+        contactEmail: props.profileState.profile.username
+      }
+    };
+  }
+
+  async send() {
+    this.setState({submitting: true, sendBugReportError: false});
+    bugReportApi().sendBugReport(this.state.bugReport).then(() => {
+      this.setState({submitting: false});
+      this.props.onClose();
+    }
+    ).catch(() => {
+      this.setState({sendBugReportError: true, submitting: false});
+    });
+
+  }
+
+  render() {
+    const {bugReport, submitting, sendBugReportError} = this.state;
+    const {onClose} = this.props;
+    return <Modal width={700} loading={submitting}>
+      <ModalTitle style={{fontWeight: 400}}>Submit a Bug</ModalTitle>
+      <ModalBody>
+        <label className='required' style={styles.fieldHeader}>Description: </label>
+        <ModalInput value={bugReport.shortDescription}
+                    onChange={(v) => this.setState({bugReport:
+                        {...bugReport, shortDescription: v.target.value}})}/>
```

Comment:
FYI, lodash has some nice utils for performing deep updates on data structures. See `_.set` and `_.update` for useful ones.